src/kudu/consensus/raft_consensus.cc

```diff
@@ -3739,10 +3739,12 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
       // TODO(mpercy): Add an API for ReadOps() to take number of ops we want,
       // instead of the max batch size.
       if (request->ops_size() > 0) {
-        RET_RESPOND_ERROR_NOT_OK(queue_->log_cache()->ReadOps(first_op_index - 1,
-                                                              max_batch_size,
-                                                              &messages,
-                                                              &preceding_id));
+        RET_RESPOND_ERROR_NOT_OK(queue_->log_cache()->ReadOps(
```

Comment:
I am not sure how this will work with 1236 mitigations. Could there be a case where middle instance has purged
more and the destination is not able to receive messages while the middle node is able to get from primary?